src/main/java/com/google/devtools/build/lib/remote/Retrier.java

```diff
@@ -141,24 +143,33 @@ public void recordSuccess() {}
 
     // 100 arbitrarily chosen as a fair balance between giving up
     // too quickly vs trying too much and giving up too late.
-    private static final int DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE = 100;
+    public static final int DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE = 100;
 
     private State state;
     private final AtomicInteger failures;
     private final AtomicInteger successes;
     private final double maxFailureRate;
     private final int minExecutionsToComputeFailureRate;
+    private final ImmutableSet<Class> ignoredErrors;
 
     public FailureRateCircuitBreaker(double maxFailureRate) {
-      this(maxFailureRate, DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE);
+      this(maxFailureRate, DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE, ImmutableSet.of());
     }
 
-    public FailureRateCircuitBreaker(double maxFailureRate, int minExecutionsToComputeFailureRate) {
+    public FailureRateCircuitBreaker(double maxFailureRate, Iterable<Class> ignoredErrors) {
+      this(maxFailureRate, DEFAULT_MIN_EXECUTIONS_TO_COMPUTE_FAILURE_RATE, ignoredErrors);
+    }
+
+    public FailureRateCircuitBreaker(
+        double maxFailureRate,
+        int minExecutionsToComputeFailureRate,
+        Iterable<Class> ignoredErrors) {
       this.maxFailureRate = maxFailureRate;
       this.state = State.ACCEPT_CALLS;
       this.failures = new AtomicInteger(0);
       this.successes = new AtomicInteger(0);
       this.minExecutionsToComputeFailureRate = minExecutionsToComputeFailureRate;
+      this.ignoredErrors = ImmutableSet.copyOf(ignoredErrors);
```

Comment:
do you foresee other errors being ignored? If not, wouldn't it be simpler to just check against `CacheNotFoundException`?

Reply:
No, not really. However it is just a couple of lines of code and makes the solution much more generic. E.g., could be used for gRPC, in case there are other exceptions to be ignored in that case.